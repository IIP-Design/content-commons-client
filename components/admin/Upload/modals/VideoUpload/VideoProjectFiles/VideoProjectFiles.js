import React, { useEffect, Fragment } from 'react';
import PropTypes from 'prop-types';
import VideoProjectFilesDesktop from './VideoProjectFileDesktop/VideoProjectFileDesktop';
import VideoProjectFilesMobile from './VideoProjectFilesMobile/VideoProjectFilesMobile';
import { VideoUploadContext } from '../VideoUpload';
import './VideoProjectFiles.scss';

const VideoProjectFiles = props => {
  // since we are using a stateless function, use a hook for mouunting/unmouting calls
  useEffect( () => {
    props.updateModalClassname( 'upload_modal prepare-files-active' );
    return () => {
      props.updateModalClassname( 'upload_modal' );
    };
  }, [] );


  return (
    <VideoUploadContext.Consumer>
      { context => {
        const message = context.files.length
          ? `Preparing ${context.files.length} files for upload...`
          : 'Please add files to upload';

        return (
          <Fragment>
            <h5 className="videoProjectFiles__headline">{ message }</h5>
            <VideoProjectFilesDesktop />
            <VideoProjectFilesMobile />
          </Fragment>
        );
      } }
    </VideoUploadContext.Consumer>
  );
};


VideoProjectFiles.propTypes = {
  updateModalClassname: PropTypes.func
};

export default VideoProjectFiles;
