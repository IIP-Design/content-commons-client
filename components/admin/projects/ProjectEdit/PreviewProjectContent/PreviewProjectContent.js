/**
 *
 * PreviewProjectContent
 *
 */

import React from 'react';
import { object, string } from 'prop-types';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import { Dropdown, Embed, Loader } from 'semantic-ui-react';

import DownloadVideo from 'components/admin/download/DownloadVideo/DownloadVideo';
import DownloadSrt from 'components/admin/download/DownloadSrt/DownloadSrt';
import DownloadThumbnail from 'components/admin/download/DownloadThumbnail/DownloadThumbnail';
import DownloadOtherFiles from 'components/admin/download/DownloadOtherFiles/DownloadOtherFiles';
import DownloadHelp from 'components/Video/DownloadHelp';

import ModalItem from 'components/modals/ModalItem/ModalItem';
import ModalContentMeta from 'components/modals/ModalContentMeta/ModalContentMeta';
import ModalDescription from 'components/modals/ModalDescription/ModalDescription';
import ModalPostMeta from 'components/modals/ModalPostMeta/ModalPostMeta';

import Notification from 'components/Notification/Notification';
import PopupTrigger from 'components/popups/PopupTrigger';
import PopupTabbed from 'components/popups/PopupTabbed';

import downloadIcon from 'static/icons/icon_download.svg';
import { getStreamData, getYouTubeId } from 'lib/utils';

import './PreviewProjectContent.scss';

/* eslint-disable react/prefer-stateless-function */
class PreviewProjectContent extends React.PureComponent {
  constructor( props ) {
    super( props );

    this.state = {
      dropDownIsOpen: false,
      selectedLanguage: 'English'
    };
  }

  getLanguages = units => (
    units.map( unit => ( {
      key: unit.language.languageCode,
      value: unit.language.displayName,
      text: unit.language.displayName
    } ) )
  )

  getProjectItems = units => (
    units.reduce( ( acc, unit ) => ( {
      ...acc,
      [unit.language.displayName]: unit
    } ), {} )
  );

  toggleArrow = () => {
    this.setState( prevState => ( {
      dropDownIsOpen: !prevState.dropDownIsOpen
    } ) );
  }

  selectLanguage = language => (
    this.setState( { selectedLanguage: language } )
  )

  handleChange = ( e, { value } ) => {
    this.toggleArrow();
    this.selectLanguage( value );
  }

  render() {
    const { error, loading, project } = this.props.data;

    if ( loading ) {
      return (
        <div
          className="preview-project-loader"
          style={ {
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            justifyContent: 'center',
            minHeight: '200px'
          } }
        >
          <Loader
            active
            inline="centered"
            style={ { marginBottom: '1em' } }
          />
          <p>Loading the project preview...</p>
        </div>
      );
    }

    if ( error ) return `Error! ${error.message}`;
    if ( !project || !Object.keys( project ).length ) return null;

    const {
      createdAt, updatedAt, projectType, team, units
    } = project;
    const { dropDownIsOpen, selectedLanguage } = this.state;

    const projectItems = this.getProjectItems( units );
    const selectedItem = projectItems[this.state.selectedLanguage];

    if ( !selectedItem || !Object.keys( selectedItem ).length ) return null;

    const {
      title, language, descPublic, files
    } = selectedItem;

    const currentUnit = files[0];
    const youTubeUrl = getStreamData( currentUnit.stream, 'youtube', 'embedUrl' );
    const vimeoUrl = getStreamData( currentUnit.stream, 'vimeo', 'embedUrl' );
    const { videoBurnedInStatus } = currentUnit;

    let thumbnailUrl = '';
    if ( selectedItem.thumbnails && selectedItem.thumbnails.length ) {
      thumbnailUrl = selectedItem.thumbnails[0].image.url;
    } else if ( project.thumbnails && project.thumbnails.length ) {
      thumbnailUrl = project.thumbnails[0].url;
    }

    const previewMsgStyles = {
      position: 'absolute',
      top: '0',
      left: '0',
      right: '0',
      // match Semantic UI border-radius
      borderTopLeftRadius: '0.28571429rem',
      borderTopRightRadius: '0.28571429rem',
      padding: '1em 1.5em',
      fontSize: '1em',
      backgroundColor: '#fdb81e'
    };

    return (
      <ModalItem
        className="project-preview"
        headline={ title }
        textDirection={ language.textDirection }
      >
        <Notification
          el="p"
          customStyles={ previewMsgStyles }
          msg={ `This is a preview of your ${projectType} project on Content Commons.` }
        />

        <div className="modal_options">
          <Dropdown
            className="modal_languages"
            value={ selectedLanguage }
            icon={ dropDownIsOpen ? 'chevron up' : 'chevron down' }
            options={ this.getLanguages( units ) }
            onClick={ this.toggleArrow }
            onChange={ this.handleChange }
          />

          <div className="trigger-container">
            <PopupTrigger
              toolTip="Download video"
              icon={ { img: downloadIcon, dim: 18 } }
              position="right"
              show={ projectType === 'video' }
              content={ (
                <PopupTabbed
                  title="Download this video."
                  panes={ [
                    {
                      title: 'Video File',
                      component: (
                        <DownloadVideo
                          selectedLanguageUnit={ selectedItem }
                          instructions={ `Download the video file in ${selectedLanguage}.
                            This download option is best for uploading this video to web pages.` }
                          burnedInCaptions={ videoBurnedInStatus === 'CAPTIONED' }
                        />
                      )
                    },
                    {
                      title: 'SRT',
                      component: (
                        <DownloadSrt
                          id={ this.props.id }
                          instructions="Download SRT(s)"
                        />
                      )
                    },
                    {
                      title: 'Thumbnail',
                      component: (
                        <DownloadThumbnail
                          id={ this.props.id }
                          instructions="Download Thumbnail(s)"
                        />
                      )
                    },
                    {
                      title: 'Other',
                      component: (
                        <DownloadOtherFiles
                          id={ this.props.id }
                          instructions="Download Other File(s)"
                        />
                      )
                    },
                    { title: 'Help', component: <DownloadHelp /> }
                  ] }
                />
              ) }
            />
          </div>
        </div>

        <div className="project-preview__content">
          { youTubeUrl
            && (
              <Embed
                id={ getYouTubeId( youTubeUrl ) }
                placeholder={ thumbnailUrl }
                source="youtube"
              />
            ) }

          { !youTubeUrl && vimeoUrl
            && (
              <Embed
                id={ getYouTubeId( youTubeUrl ) }
                placeholder={ thumbnailUrl }
                source="vimeo"
              />
            ) }

          <ModalContentMeta type={ projectType } dateUpdated={ updatedAt } />

          <ModalDescription description={ descPublic } />
        </div>

        <ModalPostMeta source={ team.name } datePublished={ createdAt } />
      </ModalItem>
    );
  }
}

PreviewProjectContent.propTypes = {
  id: string,
  data: object.isRequired
};

const VIDEO_PROJECT_PREVIEW_QUERY = gql`
  query VideoProjectPreview($id: ID!, $isReviewPage: Boolean!) {
    project: videoProject(id: $id) {
      id
      createdAt @skip(if: $isReviewPage)
      updatedAt @skip(if: $isReviewPage)
      projectType @skip(if: $isReviewPage)
      thumbnails @skip(if: $isReviewPage) {
        alt
        url
      }
      team @skip(if: $isReviewPage) {
        name
      }
      units {
        id
        title
        descPublic
        thumbnails {
          image {
            alt
            url
          }
        }
        language @skip(if: $isReviewPage) {
          languageCode
          displayName
          textDirection
        }
        files {
          id
          filename
          url
          filesize
          videoBurnedInStatus
          createdAt @include(if: $isReviewPage)
          duration @include(if: $isReviewPage)
          quality @include(if: $isReviewPage)
          dimensions {
            width
            height
          }
          stream {
            site
            embedUrl @skip(if: $isReviewPage)
            url @include(if: $isReviewPage)
          }
          language @include(if: $isReviewPage) {
            displayName
            textDirection
          }
        }
      }
    }
  }
`;

export default graphql( VIDEO_PROJECT_PREVIEW_QUERY, {
  options: props => ( {
    variables: {
      id: props.id,
      isReviewPage: false
    },
  } )
} )( PreviewProjectContent );

export { VIDEO_PROJECT_PREVIEW_QUERY };
