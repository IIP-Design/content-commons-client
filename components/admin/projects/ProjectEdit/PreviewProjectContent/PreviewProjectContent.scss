.project-preview {
  .modal_headline {
    margin-top: 1.5em;
  }
}

.ui.dimmer.modals {
  .preview-project-loader {
    .ui.loader:before {
      border-color: rgba(0, 0, 0, 0.1);
    }
    .ui.loader:after {
      border-color: #767676 transparent transparent;
    }
  }
}
