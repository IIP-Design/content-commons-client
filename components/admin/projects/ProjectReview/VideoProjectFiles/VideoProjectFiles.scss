@import "../../../../../styles/colors.scss";

.project_files {

  &_headline {
    position: relative;
    padding-left: 1rem;
    padding-top: 1rem;
    padding-bottom: 1rem;
    margin-right: -1rem;
    margin-left: -1rem;

    &:after,
    &:before {
      content: "";
      position: absolute;
      left: 0;  
      height: 2px;
      width: 100%;
      background-color: $light-grey;
    }

    &:after {
      bottom: 0;
    }

    &:before {
      top: 0;
    }
  }
}

 
.review-project .project_file {
  position: relative;  
  margin-top: 1em;
  @media screen and (min-width: 768px) {    
    padding-bottom: 2em;
  }

  &:after {
    content: "";
    position: absolute;
    bottom: 0;
    left: 0;  
    height: 2px;
    width: 100%;
    background-color: $light-grey;
  } 

  &_header {
    padding-bottom: 0 !important;

    &.RTL {

      .title {
        direction: rtl;
      }      
    }
  }

  &_contents {
    @media screen and (min-width: 768px) {
      padding-top: 0 !important;
    }        
  }

  &_edit {
    text-align: right;
  }

  .file_meta {
    padding-right: 0;

    img {
      width: 100%;
      margin-bottom: 0.5em;
    }        
  }

  .file_info {

    .public_description,
    .additional_keywords {
      margin: 1em 0;
    }

    .public_description {

      span {
        display: block;
      }

      &.RTL {

        span:last-child {
          direction: rtl;
        }        
      }
    }    
  }

}