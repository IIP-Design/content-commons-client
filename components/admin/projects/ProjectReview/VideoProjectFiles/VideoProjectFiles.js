/**
 *
 * VideoProjectFiles
 *
 */

import React from 'react';
import { object, string } from 'prop-types';
import Router from 'next/router';
import { graphql } from 'react-apollo';
import { Grid, Button } from 'semantic-ui-react';

import { VIDEO_PROJECT_PREVIEW_QUERY } from 'components/admin/projects/ProjectEdit/PreviewProjectContent/PreviewProjectContent';

import {
  formatBytes,
  formatDate,
  getPluralStringOrNot,
  millisToMinutesAndSeconds
} from 'lib/utils';

import './VideoProjectFiles.scss';

const VideoProjectFiles = props => {
  const { error, loading, project } = props.data;

  if ( loading ) return 'Loading the project...';
  if ( error ) return `Error! ${error.message}`;

  if ( !project || !Object.keys( project ).length ) return null;

  const { units } = project;

  const handleEdit = () => {
    const { id } = props;
    Router.push( {
      pathname: `/admin/project/unit/${id}/edit`
    } );
  };

  return (
    <section className="section section--project_files project_files layout">
      <h3 className="project_files_headline uppercase">
        { `${getPluralStringOrNot( units, 'Video' )} in Project` }
      </h3>
      { units.map( unit => (
        <div key={ unit.title } className="project_file">
          <Grid>
            <Grid.Row
              className={
                `project_file_header ${unit.files[0].language.textDirection}`
              }
            >
              <Grid.Column floated="left" mobile={ 8 }>
                <h4 className="title">{ unit.title }</h4>
              </Grid.Column>
              <Grid.Column floated="right" mobile={ 8 } className="project_file_edit">
                <Button
                  className="project_button project_button--edit"
                  content="Edit"
                  onClick={ handleEdit }
                />
              </Grid.Column>
            </Grid.Row>

            <Grid.Row className="project_file_contents">
              <Grid.Column mobile={ 16 } tablet={ 8 } computer={ 8 } className="file_meta">
                <img src={ unit.thumbnails[0].image.url } alt={ unit.thumbnails[0].image.alt } />
                <p><b className="label">File Name:</b> { unit.files[0].filename }</p>
                <p><b className="label">Filesize:</b> { formatBytes( unit.files[0].filesize ) }</p>
                <p><b className="label">Dimensions:</b> { `${unit.files[0].dimensions.width} x ${unit.files[0].dimensions.height}` }</p>
                <p><b className="label">Uploaded:</b> { formatDate( unit.files[0].createdAt ) }</p>
                <p><b className="label">Duration:</b> { millisToMinutesAndSeconds( unit.files[0].duration, 0 ) }</p>
              </Grid.Column>

              <Grid.Column mobile={ 16 } tablet={ 8 } computer={ 8 } className="file_info">
                <p><b className="label">Language:</b> { unit.files[0].language.displayName }</p>
                <p><b className="label">Subtitles & Captions:</b> { unit.files[0].videoBurnedInStatus }</p>
                <p><b className="label">Video Type:</b> { unit.videoType }</p>
                <p><b className="label">Quality:</b> { unit.files[0].quality }</p>
                <p
                  className={
                    `public_description ${unit.files[0].language.textDirection}`
                  }
                >
                  <b className="label">PublicÂ Description:</b>
                  <span>{ unit.descPublic }</span>
                </p>
                <p><b className="label">{ unit.files[0].stream.site } URL:</b> { unit.files[0].stream.url }</p>
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </div>
      ) ) }
    </section>
  );
};

VideoProjectFiles.propTypes = {
  id: string,
  data: object
};

export default graphql( VIDEO_PROJECT_PREVIEW_QUERY, {
  options: props => ( {
    variables: {
      id: props.id,
      isReviewPage: true
    },
  } )
} )( VideoProjectFiles );
