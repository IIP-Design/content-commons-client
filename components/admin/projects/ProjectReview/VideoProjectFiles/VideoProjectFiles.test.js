import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import Router from 'next/router';
import { MockedProvider } from 'react-apollo/test-utils';
import { Icon, Loader } from 'semantic-ui-react';
import { VIDEO_PROJECT_PREVIEW_QUERY } from 'components/admin/projects/ProjectEdit/PreviewProjectContent/PreviewProjectContent';
import VideoProjectFiles from './VideoProjectFiles';

const props = { id: '123' };

const mocks = [
  {
    request: {
      query: VIDEO_PROJECT_PREVIEW_QUERY,
      variables: { id: props.id, isReviewPage: true }
    },
    result: {
      data: {
        project: {
          id: '123',
          units: [
            {
              id: 'eng931',
              title: 'Made in America',
              descPublic: 'The value and meaning of the words “Made in America” come from a rich history of innovation and perseverance. Stay for a brief history of manufacturing in the USA and hear some of the positive impacts it has brought to the world and how its benefits are felt today.',
              thumbnails: [
                {
                  image: {
                    alt: 'A man wearing a hardhat walks through an empty factory.',
                    url: 'https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg'
                  }
                }
              ],
              files: [
                {
                  id: 'eng183',
                  filename: 'madeinamerica_english.mp4',
                  url: 'https://video-download-url.com',
                  use: {
                    name: 'Full Video'
                  },
                  filesize: 662595174,
                  videoBurnedInStatus: 'CLEAN',
                  createdAt: '2019-03-20T15:09:24.975Z',
                  updatedAt: '2019-04-02T16:28:31.888Z',
                  quality: 'WEB',
                  duration: 556000,
                  dimensions: {
                    width: 1920,
                    height: 1080
                  },
                  stream: [
                    {
                      site: 'YouTube',
                      url: 'https://www.youtube.com/watch?1evw4fRu3bo',
                      embedUrl: 'https://www.youtube.com/embed/1evw4fRu3bo'
                    },
                    {
                      site: 'Vimeo',
                      url: 'https://vimeo.com/827301171',
                      embedUrl: 'https://player.vimeo.com/video/827301171'
                    }
                  ],
                  language: {
                    displayName: 'English',
                    textDirection: 'LTR'
                  }
                }
              ]
            },
            {
              id: 'fr0381',
              title: 'Fabriqué en Amérique',
              descPublic: 'La valeur et le sens de l’expression Fabriqué en Amérique, ont leur origine dans un passé riche d’innovation et de persévérance. Découvrez une partie de l’histoire de l’industrie manufacturière aux États-Unis et l’influence positive qu’elle a eue sur le monde, dans le passé et encore aujourd’hui.',
              thumbnails: [
                {
                  image: {
                    alt: 'A man wearing a hardhat walks through an empty factory.',
                    url: 'https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg'
                  }
                }
              ],
              files: [
                {
                  id: 'fr1911',
                  filename: 'madeinamerica_french.mp4',
                  url: 'https://video-download-url.com',
                  use: {
                    name: 'Full Video'
                  },
                  filesize: 662595174,
                  videoBurnedInStatus: 'CLEAN',
                  createdAt: '2019-03-20T15:09:24.975Z',
                  updatedAt: '2019-04-02T16:28:31.888Z',
                  quality: 'WEB',
                  duration: 556000,
                  dimensions: {
                    width: 1920,
                    height: 1080
                  },
                  stream: [
                    {
                      site: 'Vimeo',
                      url: 'https://vimeo.com/827301171',
                      embedUrl: 'https://player.vimeo.com/video/827301171'
                    }
                  ],
                  language: {
                    displayName: 'French',
                    textDirection: 'LTR'
                  },
                }
              ]
            }
          ]
        }
      }
    }
  }
];

const Component = (
  <MockedProvider mocks={ mocks } addTypename={ false }>
    <VideoProjectFiles { ...props } />
  </MockedProvider>
);

describe( '<VideoProjectFiles />', () => {
  it( 'renders initial loading state without crashing', () => {
    const wrapper = mount( Component );
    const videoProjectFiles = wrapper.find( 'VideoProjectFiles' );
    const loader = (
      <Loader
        active
        inline="centered"
        style={ { marginBottom: '1em' } }
        content="Loading project file(s)..."
      />
    );

    expect( videoProjectFiles.exists() ).toEqual( true );
    expect( videoProjectFiles.contains( loader ) ).toEqual( true );
    expect( toJSON( videoProjectFiles ) ).toMatchSnapshot();
  } );

  it( 'renders error message & icon if error is thrown', async () => {
    const errorMocks = [
      {
        request: {
          query: VIDEO_PROJECT_PREVIEW_QUERY,
          variables: { id: props.id, isReviewPage: true }
        },
        result: {
          errors: [{ message: 'There was an error.' }]
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={ errorMocks }>
        <VideoProjectFiles { ...props } />
      </MockedProvider>
    );
    // wait for the data and !loading
    await wait( 0 );
    wrapper.update();

    const videoProjectFiles = wrapper.find( 'VideoProjectFiles' );
    const div = videoProjectFiles
      .find( 'div.video-project-files.error' );
    const icon = <Icon color="red" name="exclamation triangle" />;
    const span = <span>Loading error...</span>;

    expect( div.exists() ).toEqual( true );
    expect( videoProjectFiles.contains( icon ) )
      .toEqual( true );
    expect( videoProjectFiles.contains( span ) )
      .toEqual( true );
  } );

  it( 'renders `null` if project is `null`', async () => {
    const nullMocks = [
      {
        request: {
          query: VIDEO_PROJECT_PREVIEW_QUERY,
          variables: { id: props.id, isReviewPage: true }
        },
        result: {
          data: { project: null }
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={ nullMocks }>
        <VideoProjectFiles { ...props } />
      </MockedProvider>
    );
    await wait( 0 );
    wrapper.update();

    const videoProjectFiles = wrapper.find( 'VideoProjectFiles' );

    expect( videoProjectFiles.html() ).toEqual( null );
  } );

  it( 'renders the final state', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const videoProjectFiles = wrapper.find( 'VideoProjectFiles' );

    expect( toJSON( videoProjectFiles ) ).toMatchSnapshot();
  } );

  it( 'clicking the Edit button redirects to <VideoEdit />', async () => {
    const wrapper = mount( Component );
    await wait( 0 );
    wrapper.update();

    const videoProjectFiles = wrapper.find( 'VideoProjectFiles' );
    const header = videoProjectFiles.find( '.project_file_header' );
    const editBtns = header.find( 'Button.project_button--edit' );
    Router.push = jest.fn();

    editBtns.forEach( btn => {
      btn.simulate( 'click' );
      expect( Router.push ).toHaveBeenCalledWith( {
        pathname: `/admin/project/video/${props.id}/edit`
      } );
    } );
  } );
} );
