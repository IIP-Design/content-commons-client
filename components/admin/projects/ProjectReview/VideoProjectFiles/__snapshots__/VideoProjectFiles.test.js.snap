// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoProjectFiles /> renders initial loading state without crashing 1`] = `
<VideoProjectFiles
  data={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": true,
      "networkStatus": 1,
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {
        "id": "123",
        "isReviewPage": true,
      },
    }
  }
  id="123"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": "200px",
      }
    }
  >
    <Loader
      active={true}
      content="Loading project file(s)..."
      inline="centered"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      <div
        className="ui active text centered inline loader"
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Loading project file(s)...
      </div>
    </Loader>
  </div>
</VideoProjectFiles>
`;

exports[`<VideoProjectFiles /> renders the final state 1`] = `
<VideoProjectFiles
  data={
    Object {
      "error": undefined,
      "fetchMore": [Function],
      "loading": false,
      "networkStatus": 7,
      "project": Object {
        "id": "123",
        "units": Array [
          Object {
            "descPublic": "The value and meaning of the words “Made in America” come from a rich history of innovation and perseverance. Stay for a brief history of manufacturing in the USA and hear some of the positive impacts it has brought to the world and how its benefits are felt today.",
            "files": Array [
              Object {
                "createdAt": "2019-03-20T15:09:24.975Z",
                "dimensions": Object {
                  "height": 1080,
                  "width": 1920,
                },
                "duration": 556000,
                "filename": "madeinamerica_english.mp4",
                "filesize": 662595174,
                "id": "eng183",
                "language": Object {
                  "displayName": "English",
                  "textDirection": "LTR",
                },
                "quality": "WEB",
                "stream": Array [
                  Object {
                    "site": "YouTube",
                    "url": "https://www.youtube.com/watch?1evw4fRu3bo",
                  },
                  Object {
                    "site": "Vimeo",
                    "url": "https://vimeo.com/827301171",
                  },
                ],
                "url": "https://video-download-url.com",
                "use": Object {
                  "name": "Full Video",
                },
                "videoBurnedInStatus": "CLEAN",
              },
            ],
            "id": "eng931",
            "thumbnails": Array [
              Object {
                "image": Object {
                  "alt": "A man wearing a hardhat walks through an empty factory.",
                  "url": "https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg",
                },
              },
            ],
            "title": "Made in America",
          },
          Object {
            "descPublic": "La valeur et le sens de l’expression Fabriqué en Amérique, ont leur origine dans un passé riche d’innovation et de persévérance. Découvrez une partie de l’histoire de l’industrie manufacturière aux États-Unis et l’influence positive qu’elle a eue sur le monde, dans le passé et encore aujourd’hui.",
            "files": Array [
              Object {
                "createdAt": "2019-03-20T15:09:24.975Z",
                "dimensions": Object {
                  "height": 1080,
                  "width": 1920,
                },
                "duration": 556000,
                "filename": "madeinamerica_french.mp4",
                "filesize": 662595174,
                "id": "fr1911",
                "language": Object {
                  "displayName": "French",
                  "textDirection": "LTR",
                },
                "quality": "WEB",
                "stream": Array [
                  Object {
                    "site": "Vimeo",
                    "url": "https://vimeo.com/827301171",
                  },
                ],
                "url": "https://video-download-url.com",
                "use": Object {
                  "name": "Full Video",
                },
                "videoBurnedInStatus": "CLEAN",
              },
            ],
            "id": "fr0381",
            "thumbnails": Array [
              Object {
                "image": Object {
                  "alt": "A man wearing a hardhat walks through an empty factory.",
                  "url": "https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg",
                },
              },
            ],
            "title": "Fabriqué en Amérique",
          },
        ],
      },
      "refetch": [Function],
      "startPolling": [Function],
      "stopPolling": [Function],
      "subscribeToMore": [Function],
      "updateQuery": [Function],
      "variables": Object {
        "id": "123",
        "isReviewPage": true,
      },
    }
  }
  id="123"
>
  <section
    className="section section--project_files project_files layout"
  >
    <h3
      className="project_files_headline uppercase"
    >
      Videos in Project
    </h3>
    <div
      className="project_file"
      key="Made in America"
    >
      <Grid>
        <div
          className="ui grid"
        >
          <GridRow
            className="project_file_header LTR"
          >
            <div
              className="row project_file_header LTR"
            >
              <GridColumn
                floated="left"
                mobile={8}
              >
                <div
                  className="left floated eight wide mobile column"
                >
                  <h4
                    className="title"
                  >
                    Made in America
                  </h4>
                </div>
              </GridColumn>
              <GridColumn
                className="project_file_edit"
                floated="right"
                mobile={8}
              >
                <div
                  className="right floated eight wide mobile column project_file_edit"
                >
                  <Button
                    as="button"
                    className="project_button project_button--edit"
                    content="Edit"
                    onClick={[Function]}
                  >
                    <button
                      className="ui button project_button project_button--edit"
                      onClick={[Function]}
                    >
                      Edit
                    </button>
                  </Button>
                </div>
              </GridColumn>
            </div>
          </GridRow>
          <GridRow
            className="project_file_contents"
          >
            <div
              className="row project_file_contents"
            >
              <GridColumn
                className="file_meta"
                computer={8}
                mobile={16}
                tablet={8}
              >
                <div
                  className="eight wide computer sixteen wide mobile eight wide tablet column file_meta"
                >
                  <img
                    alt="A man wearing a hardhat walks through an empty factory."
                    src="https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg"
                  />
                  <p>
                    <b
                      className="label"
                    >
                      File Name:
                    </b>
                     
                    madeinamerica_english.mp4
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Filesize:
                    </b>
                     
                    631.9  MB
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Dimensions:
                    </b>
                     
                    1920 x 1080
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Uploaded:
                    </b>
                     
                    March 20, 2019
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Duration:
                    </b>
                     
                    9:16
                  </p>
                </div>
              </GridColumn>
              <GridColumn
                className="file_info"
                computer={8}
                mobile={16}
                tablet={8}
              >
                <div
                  className="eight wide computer sixteen wide mobile eight wide tablet column file_info"
                >
                  <p>
                    <b
                      className="label"
                    >
                      Language:
                    </b>
                     
                    English
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Subtitles & Captions:
                    </b>
                     
                    CLEAN
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Video Type:
                    </b>
                     
                    Full Video
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Quality:
                    </b>
                     
                    WEB
                  </p>
                  <p
                    className="public_description LTR"
                  >
                    <b
                      className="label"
                    >
                      Public Description:
                    </b>
                    <span>
                      The value and meaning of the words “Made in America” come from a rich history of innovation and perseverance. Stay for a brief history of manufacturing in the USA and hear some of the positive impacts it has brought to the world and how its benefits are felt today.
                    </span>
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      YouTube
                       URL:
                    </b>
                     
                    https://www.youtube.com/watch?1evw4fRu3bo
                  </p>
                </div>
              </GridColumn>
            </div>
          </GridRow>
        </div>
      </Grid>
    </div>
    <div
      className="project_file"
      key="Fabriqué en Amérique"
    >
      <Grid>
        <div
          className="ui grid"
        >
          <GridRow
            className="project_file_header LTR"
          >
            <div
              className="row project_file_header LTR"
            >
              <GridColumn
                floated="left"
                mobile={8}
              >
                <div
                  className="left floated eight wide mobile column"
                >
                  <h4
                    className="title"
                  >
                    Fabriqué en Amérique
                  </h4>
                </div>
              </GridColumn>
              <GridColumn
                className="project_file_edit"
                floated="right"
                mobile={8}
              >
                <div
                  className="right floated eight wide mobile column project_file_edit"
                >
                  <Button
                    as="button"
                    className="project_button project_button--edit"
                    content="Edit"
                    onClick={[Function]}
                  >
                    <button
                      className="ui button project_button project_button--edit"
                      onClick={[Function]}
                    >
                      Edit
                    </button>
                  </Button>
                </div>
              </GridColumn>
            </div>
          </GridRow>
          <GridRow
            className="project_file_contents"
          >
            <div
              className="row project_file_contents"
            >
              <GridColumn
                className="file_meta"
                computer={8}
                mobile={16}
                tablet={8}
              >
                <div
                  className="eight wide computer sixteen wide mobile eight wide tablet column file_meta"
                >
                  <img
                    alt="A man wearing a hardhat walks through an empty factory."
                    src="https://staticcdp.s3.amazonaws.com/2018/05/courses.america.gov_1481/b3b38d194ff80d06dd837f57a41fe16f.jpg"
                  />
                  <p>
                    <b
                      className="label"
                    >
                      File Name:
                    </b>
                     
                    madeinamerica_french.mp4
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Filesize:
                    </b>
                     
                    631.9  MB
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Dimensions:
                    </b>
                     
                    1920 x 1080
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Uploaded:
                    </b>
                     
                    March 20, 2019
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Duration:
                    </b>
                     
                    9:16
                  </p>
                </div>
              </GridColumn>
              <GridColumn
                className="file_info"
                computer={8}
                mobile={16}
                tablet={8}
              >
                <div
                  className="eight wide computer sixteen wide mobile eight wide tablet column file_info"
                >
                  <p>
                    <b
                      className="label"
                    >
                      Language:
                    </b>
                     
                    French
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Subtitles & Captions:
                    </b>
                     
                    CLEAN
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Video Type:
                    </b>
                     
                    Full Video
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Quality:
                    </b>
                     
                    WEB
                  </p>
                  <p
                    className="public_description LTR"
                  >
                    <b
                      className="label"
                    >
                      Public Description:
                    </b>
                    <span>
                      La valeur et le sens de l’expression Fabriqué en Amérique, ont leur origine dans un passé riche d’innovation et de persévérance. Découvrez une partie de l’histoire de l’industrie manufacturière aux États-Unis et l’influence positive qu’elle a eue sur le monde, dans le passé et encore aujourd’hui.
                    </span>
                  </p>
                  <p>
                    <b
                      className="label"
                    >
                      Vimeo
                       URL:
                    </b>
                     
                    https://vimeo.com/827301171
                  </p>
                </div>
              </GridColumn>
            </div>
          </GridRow>
        </div>
      </Grid>
    </div>
  </section>
</VideoProjectFiles>
`;
